### PR TITLE
Add javadoc for eventing related functionalities

### PR DESCRIPTION
**Pull request by mmedenjak**
Adding javadoc as a preparation for https://hazelcast.atlassian.net/wiki/display/EN/Core+Part+-+Stream+Source+from+Map+and+Cache+-+Design

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/EventServiceImpl.java [hunk 1]
```diff
@@ -63,30 +63,93 @@
 import static com.hazelcast.util.FutureUtil.ExceptionHandler;
 import static com.hazelcast.util.FutureUtil.waitWithDeadline;
 
+/**
+ * Service responsible for routing and dispatching local and remote events and keeping track of listener
+ * registrations. Local events are events published on a local subscriber (the subscriber is on this node)
+ * and remote events are published on a remote subscriber (the subscriber is on a different node and the
+ * event is sent to that node). The remote events are generally asnychronous meaning that we send the event
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/EventServiceImpl.java]: They are always async (fire and forget)

File: hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/EventServiceImpl.java [hunk 2]
```diff
@@ -63,30 +63,93 @@
...
+ * and don't wait for the response. The exception to this is that every {@link #eventSyncFrequency} remote
+ * event is sent as an operation and we wait for it to be submitted to the remote queue.
+ * <p>
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/EventServiceImpl.java]: Ah ok.. Forget the above comment

**Comment by pveentjer** [comment]
Since this is just javadoc improvements, one reviewer is sufficient. So merge once green

**Comment by mmedenjak** [comment]
verify

**Comment by mmedenjak** [comment]
run-lab-run

**Pull request merged**